src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveAction.java

```diff
@@ -234,6 +252,213 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         return createSpecificRedirectResult();
     }
 
+    /**
+     * Updates comments which were edited by feedback submitter.
+     * @param commentsToUpdateId key: comment index, value: comment id
+     * @param commentsToUpdateText key: comment index, value: comment text
+     * @param responseGiverMapForComments key: comment index, value: response giver
+     * @param responseRecipientMapForComments key: comment index, value: response recipient
+     */
+    private void updateResponsesComments(Map<String, String> commentsToUpdateId,
+                                         Map<String, String> commentsToUpdateText,
+                                         Map<String, String> responseGiverMapForComments,
+                                         Map<String, String> responseRecipientMapForComments,
+                                         Map<String, String> questionIdsForComments) {
+        for (String commentIndex : commentsToUpdateId.keySet()) {
+            String commentId = commentsToUpdateId.get(commentIndex);
+            String updatedCommentText = commentsToUpdateText.get(commentIndex);
+            FeedbackResponseAttributes response =
+                    logic.getFeedbackResponse(questionIdsForComments.get(commentIndex),
+                            responseGiverMapForComments.get(commentIndex),
+                            responseRecipientMapForComments.get(commentIndex));
+            String commentGiverString =
+                    getRequestParamValue(Const.ParamsNames.COMMENT_GIVER_TYPE + commentIndex);
+            String showCommentTo = null;
+            String showGiverNameTo = null;
+            // Only instructor can set visibility settings for his/her comments.
+            if (this instanceof InstructorFeedbackSubmissionEditSaveAction) {
+                showCommentTo =
+                        getRequestParamValue(Const.ParamsNames.RESPONSE_COMMENTS_SHOWCOMMENTSTO + commentIndex);
+                showGiverNameTo =
+                        getRequestParamValue(Const.ParamsNames.RESPONSE_COMMENTS_SHOWGIVERTO + commentIndex);
+            }
+            updateResponseComment(showCommentTo, showGiverNameTo, commentId, response, updatedCommentText,
+                    commentGiverString);
+
+        }
+    }
+
+    // Updates edited comment
+    private void updateResponseComment(String showCommentTo, String showGiverNameTo, String commentId,
+            FeedbackResponseAttributes response, String updatedCommentText, String commentGiverString) {
+        FeedbackResponseCommentAttributes feedbackResponseComment = FeedbackResponseCommentAttributes
```

Comment:
`FeedbackResponseCommentAttributes` should be done in `FeedbackResponseCommentAttributes`. Let's follow the way that response does.

First extract all response that going to update/create/delete, and then call logic to do the job.